Domain/Dto/Bets.cs

```diff
@@ -10,7 +10,7 @@ public class Bets
         [StringLength(100)]
         public string Name { get; set; }
 
-        public DateTime Date { get; private set; } = DateTime.UtcNow;
+        public DateTime Date { get; private set; } = TimeProvider.System.GetUtcNow().DateTime;
```

Comment:
you will still not be able to mock the time provider in tests

Reply:
you should set the date when creating the entity in the DB via the service.
also you will need to inject the TimeProvider